bases/renku_data_services/background_jobs/core.py

```diff
@@ -67,3 +76,54 @@ async def bootstrap_user_namespaces(config: SyncConfig) -> None:
         )
         return
     await sync_user_namespaces(config)
+
+
+async def fix_mismatched_project_namespace_ids(config: SyncConfig) -> None:
+    """Fixes a problem where the project namespace relationship for projects has the wrong group ID."""
+    api_user = InternalServiceAdmin(id=ServiceAdminId.migrations)
+    authz = Authz(config.authz_config)
+    res = authz.client.ReadRelationships(
+        ReadRelationshipsRequest(
+            consistency=Consistency(fully_consistent=True),
+            relationship_filter=RelationshipFilter(
+                resource_type=ResourceType.project,
+                optional_relation=_Relation.project_namespace.value,
+                optional_subject_filter=SubjectFilter(subject_type=ResourceType.group.value),
+            ),
+        )
+    )
+    async for rel in res:
+        logging.info(f"Checking project namespace - group relation {rel} for correct group ID")
+        project_id = rel.relationship.resource.object_id
+        project = await config.project_repo.get_project(api_user, project_id)
+        if project.namespace.kind != NamespaceKind.group:
+            continue
+        correct_group_id = project.namespace.underlying_resource_id
+        authzed_group_id = rel.relationship.subject.object.object_id
+        if authzed_group_id != correct_group_id:
+            logging.info(
+                f"The project namespace ID in Authzed {authzed_group_id} "
+                f"does not match the expected group ID {correct_group_id}, correcting it..."
+            )
```

Comment:
Logging doesn't work here btw (see #267).

Reply:
I know I hope that it will when we fix the logging setup.

Reply:
I will try swapping the alembic config level to info and see what happens. Hopefully alembic does not flood the console with lots of random messages.

Reply:
Logs look normal with the level set to INFO in alembic. So I will change that.

Reply:
An easy way to have the logging work in the background job is to use `logger = logging.getLogger("background_jobs")` and use that.

Reply:
```python
logger = logging.getLogger("background_jobs")
logger.setLevel(logging.INFO)
```

Reply:
With logging level set to info in alembic it works, so I will keep it as is.